packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -0,0 +1,247 @@
+   */
+  public static exact(pathMatch: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePathSpecificMatch({ exact: pathMatch }, exactPathRewrite);
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   * @param regexMatch The regex used to match the path.
+   */
+  public static regex(regexMatch: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePathSpecificMatch({ regex: regexMatch }, exactPathRewrite);
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixMatch This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   *  You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   *  and you want the gateway route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   */
+  public static startingWith(prefixMatch: string, prefixRewrite?: HttpGatewayRoutePrefixPathRewrite): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixMatchImpl(prefixMatch, prefixRewrite);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixMatchImpl extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly prefixPathRewrite?: HttpGatewayRoutePrefixPathRewrite,
+  ) {
+    super();
+  }
+
+  bind(scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.prefixPathRewrite?.bind(scope).prefixPathPath,
+    };
+  }
+}
+
+class HttpGatewayRoutePathSpecificMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly match: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly exactPathRewrite?: string,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.match,
+      wholePathRewrite: {
+        exact: this.exactPathRewrite,
+      },
+    };
+  }
+}
+
+/**
+ * Prefix properties for HTTP gateway route rewrite.
+ */
+export interface HttpGatewayRoutePrefixPathRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefixPathPath?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Used to generate HTTP gateway route rewrite other than the host name.
+ */
+export abstract class HttpGatewayRoutePrefixPathRewrite {
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   * When enabled, rewrites the matched prefix in Gateway Route to '/'.
+   * When disabled, retains the original prefix from the request.
```

Comment:
Can you explain how is this different than not specifying a rewrite at all?

Reply:
When `path` rewrite is not specified, `defaultPrefix` is `ENABLED`. This means `prefix` path is re-written to `/`.